File: problems/2078-two-furthest-houses-with-different-colors/solution.cpp
class Solution {
public:
    int maxDistance(vector<int>& colors) {
        int n = colors.size();
        int max_dist = 0;
        for (int i = 0; i < n; ++i) {
            for (int j = n - 1; j > i; --j) {
                if (colors[i] != colors[j]) {
                    max_dist = std::max(max_dist, j - i);
                    break;
                }
            }
        }
        return max_dist;
    }
};

File: problems/2078-two-furthest-houses-with-different-colors/solution.java
class Solution {
    public int maxDistance(int[] colors) {
        int n = colors.length;
        int maxDist = 0;

        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                if (colors[i] != colors[j]) {
                    maxDist = Math.max(maxDist, Math.abs(i - j));
                }
            }
        }

        return maxDist;
    }
}